packages/pods-core/src/builtin/GDocPod.ts

```diff
@@ -0,0 +1,273 @@
+import { ImportPod, ImportPodConfig, ImportPodPlantOpts } from "../basev3";
+import { JSONSchemaType } from "ajv";
+import { PodUtils } from "../utils";
+import axios from "axios";
+import { googleDocsToMarkdown } from "docs-markdown";
+import _ from "lodash";
+import {
+  DendronError,
+  DVault,
+  NoteProps,
+  NoteUtils,
+  stringifyError,
+  DEngineClient,
+} from "@dendronhq/common-all";
+
+const ID = "dendron.gdoc";
+
+type GDocImportPodCustomOpts = {
+  /**
+   * google docs personal access token
+   */
+  token: string;
+
+   /**
+   * document Id of doc to import
+   */
+  documentId: string;
+
+  /**
+   * name of hierarchy to import into
+   */
+  fname: string;
+
+   /**
+   * import comments from the doc in text or json format
+   */
+  importComments? : ImportComments;
+};
+
+type ImportComments = {
+  enable: boolean;
+  format?: string;
+}
+
+type GDocImportPodConfig = ImportPodConfig & GDocImportPodCustomOpts;
+
+export type GDocImportPodPlantOpts = ImportPodPlantOpts;
+
+export class GDocImportPod extends ImportPod<GDocImportPodConfig> {
+  static id: string = ID;
+  static description: string = "import google doc";
+
+  get config(): JSONSchemaType<GDocImportPodConfig> {
+    return PodUtils.createImportConfig({
+      required: ["token", "fname", "documentId"],
+      properties: {
+        token: {
+          type: "string",
+          description: "google docs personal access token",
+        },
+        documentId: {
+          type: "string",
+          description: "document Id of doc to import",
+        },
+        fname: {
+          type: "string",
+          description: "name of hierarchy to import into",
+        },
+        importComments: {
+          type: "object",
+          nullable: true,
+          description: "import comments from the doc in text or json format",
+          required : ["enable"],
+          properties: {
+            enable: {
+              type: "boolean",
+              default: "false"
+            },
+            format: {
+              type: "string",
+              enum: ["json","text"],
+              default: "json",
+              nullable: true,
+
+            }
+          }
+        }
+      },
+    }) as JSONSchemaType<GDocImportPodConfig>;
+  }
+
+  /*
+   * method to get data from google document
+   */
+  getDataFromGDoc = async (opts: Partial<GDocImportPodConfig>, 
+    config: ImportPodConfig
+    ): Promise<Partial<NoteProps>>  => {
+    let response: Partial<NoteProps>;
+    const {documentId, token, fname } = opts;
+    const headers = {
+      Authorization: `Bearer ${token}`,
+      "Content-Type": "application/json",
+    };
+    try{
+      const result = await axios.get(
+        `https://docs.googleapis.com/v1/documents/${documentId}`,
+        { headers }
+      )
+      const markdown = googleDocsToMarkdown(result.data);
+      const body= markdown.split("---")
```

Comment:
What if the doc contents naturally has a `---`?

Reply:
Hm I must look for this scenario, since the package that we have used to convert docs to .md returns a string with some metadata wrapped around `---`

Reply:
Yes I should consider this case, will fix this in new commit.